cloud/services/compute/vm.go

```diff
@@ -0,0 +1,212 @@
+package compute
+
+import (
+	b64 "encoding/base64"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+
+	"github.com/benbjohnson/clock"
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/scope"
+	"net"
+	"time"
+
+	tag "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	osc "github.com/outscale/osc-sdk-go/v2"
+)
+
+//go:generate ../../../bin/mockgen -destination mock_compute/vm_mock.go -package mock_compute -source ./vm.go
+type OscVmInterface interface {
+	CreateVm(machineScope *scope.MachineScope, spec *infrastructurev1beta1.OscVm, subnetId string, securityGroupIds []string, privateIps []string, vmName string) (*osc.Vm, error)
+	DeleteVm(vmId string) error
+	GetVm(vmId string) (*osc.Vm, error)
+	GetVmState(vmId string) (string, error)
+	CheckVmState(clockInsideLoop time.Duration, clockLoop time.Duration, state string, vmId string) error
+}
+
+// ValidateKeypairName check that KeypairName is a valid name of keypair
+func ValidateKeypairName(keypairName string) (string, error) {
+	isValidateKeypairName := regexp.MustCompile("^[\x20-\x7E]{0,255}$").MatchString
+	if isValidateKeypairName(keypairName) {
+		return keypairName, nil
+	} else {
+		return keypairName, errors.New("Invalid KeypairName")
+	}
+}
+
+// ValidateImageId check that imageId is a valid imageId
+func ValidateImageId(imageId string) (string, error) {
+	switch {
+	case strings.Contains(imageId, "ami"):
+		return imageId, nil
+	default:
+		return imageId, errors.New("Invalid imageId")
+	}
+}
+
+// ValidateDeviceName check that DeviceName  is a valid DeviceName
+func ValidateDeviceName(deviceName string) (string, error) {
+	switch {
+	case strings.Contains(deviceName, "/dev/xvd") && len(deviceName) == 9:
```

Comment:
Maybe check that it starts with '/dev/xvd'  and last character is a number ?

Reply:
last character is a number is never a number ?

Reply:
letter sorry :)

Reply:
Ok